test/testactivitydata.cpp

```diff
@@ -0,0 +1,143 @@
+        });
+
+        QJsonObject testData({
+            {QStringLiteral("object_type"), objectType},
+            {QStringLiteral("activity_id"), activityId},
+            {QStringLiteral("type"), activityType},
+            {QStringLiteral("subject"), subject},
+            {QStringLiteral("message"), message},
+            {QStringLiteral("object_name"), objectName},
+            {QStringLiteral("link"), link},
+            {QStringLiteral("datetime"), datetime},
+            {QStringLiteral("icon"), icon},
+            {QStringLiteral("subject_rich"), subjectRichData},
+            {QStringLiteral("previews"), QJsonArray({previewData})},
+        });
+
+        OCC::Activity activity = OCC::Activity::fromActivityJson(testData, account);
+        QCOMPARE(activity._type, OCC::Activity::ActivityType);
+        QCOMPARE(activity._objectType, objectType);
+        QCOMPARE(activity._id, activityId);
+        QCOMPARE(activity._fileAction, activityType);
+        QCOMPARE(activity._accName, account->displayName());
+        QCOMPARE(activity._subject, subject);
+        QCOMPARE(activity._message, message);
+        QCOMPARE(activity._file, objectName);
+        QCOMPARE(activity._link, link);
+        QCOMPARE(activity._dateTime, QDateTime::fromString(datetime, Qt::ISODate));
+
+        QCOMPARE(activity._subjectRichParameters.count(), 1);
+        QCOMPARE(activity._subjectDisplay, QStringLiteral("You created ").append(fileName));
+
+        QCOMPARE(activity._previews.count(), 1);
+        // We want the different icon when we have a preview
+        QCOMPARE(activity._icon, QStringLiteral("qrc:///client/theme/colored/add-bordered.svg"));
+
+        if(fileFormat == "txt") {
+            QCOMPARE(activity._previews[0]._source, account->url().toString().append(QStringLiteral("/index.php/apps/theming/img/core/filetypes/text.svg")));
+            QCOMPARE(activity._previews[0]._isMimeTypeIcon, true);
+            QCOMPARE(activity._previews[0]._mimeType, mimeType);
+        } else if(fileFormat == "pdf") {
+            QCOMPARE(activity._previews[0]._source, account->url().toString().append(QStringLiteral("/index.php/apps/theming/img/core/filetypes/application-pdf.svg")));
+            QCOMPARE(activity._previews[0]._isMimeTypeIcon, true);
+        } else {
+            QCOMPARE(activity._previews[0]._source, previewUrl);
+            QCOMPARE(activity._previews[0]._isMimeTypeIcon, false);
+        }
+
+        QCOMPARE(activity._previews[0]._mimeType, mimeType);
+    }
+
+    QScopedPointer<FakeQNAM> fakeQnam;
+    OCC::AccountPtr account;
+
+private slots:
+    void initTestCase()
+    {
+        account = OCC::Account::create();
+        account->setCredentials(new FakeCredentials{fakeQnam.data()});
+        account->setUrl(QUrl(("http://example.de")));
+        auto *cred = new HttpCredentialsTest("testuser", "secret");
+        account->setCredentials(cred);
+    }
+
+    void testFromJson()
```

Comment:
can you make this test a data driven one ?
please have a look at this tutorial from [Qt doc](https://doc.qt.io/qt-5/qttestlib-tutorial2-example.html)

Reply:
I have used the column and row structure from the documentation now, thanks